MovingBlocks/Terasology
# feat(extract-subsystems): extract `Context`, `@In`, `InjectionHelper` and `@Share` to Subsystem

### Contains

Part of #4304.

Provide possible to Injectable classes(`@Share`, `@In` and `Context` usage) with another subsystems.

Should be replaced with gestalt v8 (di)

### How to test

1.Run your favorite game (hint: Terasology)
2. Check it don't fail whit null pointer exception at `@In` field.

### Outstanding before merging
